libibverbs/device.c

```diff
@@ -262,6 +267,8 @@ struct ibv_context *__ibv_open_device(struct ibv_device *device)
 	context->cmd_fd = cmd_fd;
 	pthread_mutex_init(&context->mutex, NULL);
 
+	ibverbs_device_hold(device);
+
```

Comment:
We already agreed that TS should be used to detect dropped devices as part of get device list flow.

This discussion is for further cases which are not directly related to this series and can happen also today, we prefer going ahead and take the series, any further handling can be done on top of this series.